programs/mango-v4/src/instructions/token_deposit.rs

```diff
@@ -100,18 +100,30 @@ impl<'a, 'info> DepositCommon<'a, 'info> {
         &self,
         remaining_accounts: &[AccountInfo],
         amount: u64,
+        reduce_only: bool,
         allow_token_account_closure: bool,
     ) -> Result<()> {
         require_msg!(amount > 0, "deposit amount must be positive");
 
         let token_index = self.bank.load()?.token_index;
 
+        let bank = self.bank.load()?;
+
         // Get the account's position for that token index
         let mut account = self.account.load_mut()?;
 
         let (position, raw_token_index) = account.token_position_mut(token_index)?;
 
-        let amount_i80f48 = I80F48::from(amount);
+        let amount_i80f48 = if reduce_only || bank.is_reduce_only() {
+            position
+                .native(&bank)
+                .min(I80F48::ZERO)
+                .abs()
+                .min(I80F48::from(amount))
```

Comment:
same "error vs silent reduction" thing as for perps

Reply:
done